java/statestore-commit/src/main/java/sleeper/commit/StateStoreCommitRequestDeserialiser.java

```diff
@@ -88,15 +109,15 @@ public StateStoreCommitRequest deserialize(JsonElement json, Type wrapperType, J
                 case INGEST_ADD_FILES:
                     return StateStoreCommitRequest.forIngestAddFiles(
                             context.deserialize(requestObj, IngestAddFilesCommitRequest.class));
-                case STORED_IN_S3:
-                    return StateStoreCommitRequest.storedInS3(
-                            context.deserialize(requestObj, StateStoreCommitRequestInS3.class));
                 case COMPACTION_JOB_ID_ASSIGNMENT:
                     return StateStoreCommitRequest.forCompactionJobIdAssignment(
                             context.deserialize(requestObj, CompactionJobIdAssignmentCommitRequest.class));
                 case SPLIT_PARTITION:
                     return StateStoreCommitRequest.forSplitPartition(
                             context.deserialize(requestObj, SplitPartitionCommitRequest.class));
+                case STORED_IN_S3:
```

Comment:
I felt like the stored in S3 type should be separate from the others because it's handled differently. I'm not sure there's a good way to make this clear though.